### PR TITLE
json: move JSON keys to PascalCase

### PR DESCRIPTION
**Pull request by behrmann**
Generally, the keys are changed to whatever the format in the config files is, which is a nice side effect.

### CODE REVIEW

File: mkosi/config.py [hunk 1]
```diff
@@ -684,7 +685,13 @@ def from_namespace(cls, ns: argparse.Namespace) -> "MkosiArgs":
 
     def to_json(self, *, indent: Optional[int] = 4, sort_keys: bool = True) -> str:
         """Dump MKosiArgs as JSON string."""
-        return json.dumps(dataclasses.asdict(self), cls=MkosiJsonEncoder, indent=indent, sort_keys=sort_keys)
+        def key_transformer(k: str) -> str:
+            if (s := SETTINGS_LOOKUP_BY_DEST.get(k)) is not None:
+                return s.name
+            return "".join(p.capitalize() for p in k.split("_"))
```
Comment (DaanDeMeyer) [inline mkosi/config.py:691]: Does this make sense for `MkosiArgs`? It won't have anything in the settings

**Review by DaanDeMeyer — CHANGES_REQUESTED** [review]

File: mkosi/config.py (hunk 1 above)
Comment (behrmann) [reply mkosi/config.py:691]: Without the tests fail because of `AutoBump`

Comment (DaanDeMeyer) [reply mkosi/config.py:691]: I'm confused, we don't have an `AutoBump` setting in `SETTINGS` right?

Comment (behrmann) [reply mkosi/config.py]: Ah, misunderstood you and thought you meant the last line of the hunk. I thought the `genkey_*` keys were in `SETTINGS`, but that was me confusing the long block of parser settings with the `SETTINGS` one...

File: mkosi/config.py [hunk 2]
```diff
@@ -936,7 +946,13 @@ def cache_manifest(self) -> dict[str, Any]:
 
     def to_json(self, *, indent: Optional[int] = 4, sort_keys: bool = True) -> str:
         """Dump MKosiConfig as JSON string."""
-        return json.dumps(dataclasses.asdict(self), cls=MkosiJsonEncoder, indent=indent, sort_keys=sort_keys)
+        def key_transformer(k: str) -> str:
+            if (s := SETTINGS_LOOKUP_BY_DEST.get(k)) is not None:
+                return s.name
+            return "".join(p.capitalize() for p in k.split("_"))
```
Comment (DaanDeMeyer) [inline mkosi/config.py:952]: Do we need these fallbacks at all for `MkosiConfig`? I think everything in there has an associated setting no?

File: mkosi/config.py [hunk 3]
```diff
@@ -950,8 +966,13 @@ def _load_json(cls, s: Union[str, dict[str, Any], SupportsRead[str], SupportsRea
         else:
             raise ValueError(f"{cls.__name__} can only be constructed from JSON from strings, dictionaries and files.")
 
-        transformer = json_type_transformer(cls)
-        return {k: transformer(k, v) for k, v in j.items()}
+        value_transformer = json_type_transformer(cls)
+        def key_transformer(k: str) -> str:
+            if (s := SETTINGS_LOOKUP_BY_NAME.get(k)) is not None:
+                return s.dest
+            return "_".join(part.lower() for part in FALLBACK_NAME_TO_DEST_SPLITTER.split(k))
```
Comment (DaanDeMeyer) [inline mkosi/config.py:973]: Ditto, do we need the fallback for `MkosiConfig`?

**Review by DaanDeMeyer — CHANGES_REQUESTED** [review]

File: mkosi/config.py (hunk 2 above)
Comment (behrmann) [reply mkosi/config.py:952]: Yes, `preset` is an attribute of `MkosiConfig`, but there is no accompanying `MkosiConfigSetting`.

File: mkosi/config.py (hunk 3 above)
Comment (behrmann) [reply mkosi/config.py:973]: Same, `preset` is the culprit.

**Review by DaanDeMeyer — APPROVED** [review]

**Pull request merged**